sw/host/opentitantool/src/command/ecdsa.rs

```diff
@@ -0,0 +1,264 @@
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        let private_key = EcdsaPrivateKey::new();
+        let mut der_file = self.output_dir.to_owned();
+        der_file.push(&self.basename);
+        der_file.set_extension("der");
+        private_key.save(&der_file)?;
+
+        let public_key = private_key.public_key();
+        der_file.set_extension("pub.der");
+        public_key.save(&der_file)?;
+        Ok(None)
+    }
+}
+
+/// Export public information of an ECDSA key
+/// to a C header that can be used in the ROM or ROM_EXT
+#[derive(Debug, Args)]
+pub struct EcdsaKeyExportCommand {
+    /// RSA public or private key file in DER format.
+    der_file: PathBuf,
+    /// output header file to generate.
+    output_file: Option<PathBuf>,
+}
+
+impl CommandDispatch for EcdsaKeyExportCommand {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        _transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Annotate>>> {
+        let key = load_pub_or_priv_key(&self.der_file)?;
+        let key = key.to_raw();
+
+        let output_path = match &self.output_file {
+            Some(path) => path.clone(),
+            None => {
+                // if no output file is provided, derive the name of the file from the
+                // name of the key
+                let mut out_path = self.der_file.clone();
+                out_path.set_extension("h");
+                out_path
+            }
+        };
+        // We try to catch the mistake of a user that specifies the key file as output,
+        // which would overwrite it. This will not detect situations where there is a symlink
+        // involved so this will only catch "obvious" mistakes.
+        if self.der_file == output_path {
+            bail!("the output file is the same as the key file, this would overwrite the key, not allowing this")
+        }
+        println!("exporting key to {}", output_path.display());
+
+        let mut file = File::create(&output_path)?;
+        // construct a key name from the key file name
+        let keyname = self
+            .der_file
+            .file_name()
+            .expect("this should be a valid file name since we opened the file")
+            .to_string_lossy();
+        let re = Regex::new(r#"(.pub)?.der$"#).unwrap();
+        let keyname = re.replace_all(&keyname, "");
+        let re = Regex::new(r#"[^a-zA-Z0-9]"#).unwrap();
+        let keyname = re.replace_all(&keyname, "_").to_ascii_uppercase();
+        // we cannot know the purpose of this key but the header guard should probably include it
+        // so we add some extra text to the guard that will not compile to force the user to fix it
+        let header_guard = format!("{}_H", keyname);
+        // write header guard
+        writeln!(&mut file, "#ifndef {}", header_guard)?;
+        writeln!(&mut file, "#define {}", header_guard)?;
+        writeln!(&mut file)?;
+        writeln!(&mut file, "#define {} \\", keyname)?;
```

Comment:
What is the preferred representation of the ECDSA key in firmware?

Tentatively, I've expressed the key as the X and Y points as little-endian 32-bit words.

Reply:
This is how the cryptolib represents it, so I think it's the best bet.